src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -890,21 +890,35 @@ void PasswordSafeFrame::RefreshToolbarButtons()
   if (toolbar->GetToolCount() > 0) {
     toolbar->ClearTools();
   }
+  bool toolbarShowText = pref->GetPref(PWSprefs::ToolbarShowText);
+  bool showMenuSeprator = pref->GetPref(PWSprefs::ShowMenuSeparator);
 
   for (const auto & PwsToolbarButton : PwsToolbarButtons) {
-    if (pref->GetPref(PWSprefs::ShowMenuSeparator) && (PwsToolbarButton.IsSeparator())) {
-      toolbar->AddSeparator()->SetId(PwsToolbarButton.id);
+    if (PwsToolbarButton.IsSeparator()) {
+      if (showMenuSeprator)
```

Comment:
Why not combine the two if conditions?

Reply:
Because of the logic, the following else depends only from type of entry. In case of combination an assert is the result of execution in case of configuration to not showing separation.